bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/link/ItemChannelLinkResource.java

```diff
@@ -314,4 +319,45 @@ private boolean isEditable(String linkId) {
         }
         return thing.getChannel(channelUID);
     }
+
+    @GET
+    @Path("/orphans")
+    @Operation(operationId = "getOrphanLinks", summary = "Get orphan links between items and broken/non-existent thing channels", responses = {
+            @ApiResponse(responseCode = "200", description = "List of broken links") })
+    public Response getOrphanLinks() {
+        return Response.ok(getOrphanInternalLinks()).build();
+    }
+
+    public List<BrokenItemChannelLinkDTO> getOrphanInternalLinks() {
+        Collection<Item> items = itemRegistry.getItems();
+        Collection<Thing> things = thingRegistry.getAll();
+        Collection<ItemChannelLink> itemChannelLinks = itemChannelLinkRegistry.getAll();
+
+        List<BrokenItemChannelLinkDTO> results = new ArrayList<>();
+
+        Collection<ChannelUID> channelUIDS = things.stream().map(Thing::getChannels).flatMap(List::stream)
+                .map(Channel::getUID).collect(Collectors.toSet());
+        Collection<String> itemNames = items.stream().map(Item::getName).collect(Collectors.toSet());
+
+        itemChannelLinks.forEach(itemChannelLink -> {
+            if (!channelUIDS.contains(itemChannelLink.getLinkedUID())) {
+                boolean isEditable = isEditable(itemChannelLink.getUID());
+
+                if (!itemNames.contains(itemChannelLink.getItemName())) {
+                    results.add(new BrokenItemChannelLinkDTO(itemChannelLink,
+                            BrokenItemChannelLinkDTO.ItemChannelLinkProblem.ITEM_AND_THING_CHANNEL_MISSING,
+                            isEditable));
+                } else {
+                    results.add(new BrokenItemChannelLinkDTO(itemChannelLink,
+                            BrokenItemChannelLinkDTO.ItemChannelLinkProblem.THING_CHANNEL_MISSING, isEditable));
```

Comment:
For returning the data in the REST API you should probably map the `ItemChannelLink` to an `EnrichedItemChannelLinkDTO`which already includes the `editable` property.

Reply:
Will update MainUI PR once this one merged